wg-naming/language-evaluation-framework.md

```diff
@@ -0,0 +1,111 @@
+
+### First-order concerns
+
+First-order concerns are characterized by: 
+
+- Overtness: regardless of its use in the context of code or technology, there is little to no ambiguity outside of technology as to whether the language in question indicates harm.
+- Identity-specificity: language in question specifically unambiguously identifies a group of people.
+
+#### Is the term overtly racist?
+
+Examples include “master/slave”.
+
+#### Is the term overtly sexist, transphobic, or pejorative about a gender identity?
+
+Examples do _not_ include “transclusion” of dependencies, or “binary” operators. 
+
+#### Is the term overtly ableist, or pejorative to neurodiverse or disabled people
+
+Examples include performing “sanity checks”. 
+
+#### Is the term overtly homophobic? 
+
+Examples do not include “homogenizing” or “homogenous” data. 
+
+### Second-order concerns
+
+Second-order concerns are characterized by: 
+
+- Ambiguity: outside the context of code or technology, language might have connotations related to harmful scenarios like war, militarization, or policing, but the actual etymology of the term is not related to harm of a specific identity
+- Lack of identity-specificity: concerns in this category do not target specific identities, or do so in a non-overt way.
+
+#### Is the term violent?
+
+Examples include “KILL” commands in Unix systems.
+
+#### Is the term militaristic?
+
+Examples include “marshal/unmarshal”.
+
+
+### Third-order concerns
+
+Third-order concerns are characterized by:
+
+- Language focus: is the language in use a metaphor that could be described more precisely using different words?
+
+
+#### Is the term evocative instead of descriptive?
+
+Examples include “PetSet” (evocative) versus “StatefulSet” (descriptive). 
+
+#### Is the term ambiguous?
+
+Examples include the use of ABORT/STOP/KILL in Unix-like systems, where they map to specific behaviors, versus general usage in programming languages, where they map to different behaviors or are used interchangeably. 
+
+## Footnotes 
+
+### The element of time
+
+In general, strong democratic societies become more progressive and accepting as time passes. This is a feature, not a bug. 
+
+The result of this, for your work, is that terms that were once deemed acceptable may, at some future point, be deemed unacceptable. 
+
+We recommend:
+
+- Placing a date at the top of any documents/recommendations related to naming, language inclusivity, or harm reduction 
+- Expecting that some of your work will need re-evaluation at a later date
+- An openness to reversing decisions 
+
+
+### Dealing with trolls
+
+In the handful of months since this work began, both Kubernetes as a whole and WG Naming have dealt with a number of issues and comments from trolls. We anticipate anyone using this document to guide their own work will receive the same kind of attention. 
```

Comment:
```suggestion
In the handful of months since this work began, both Kubernetes as a whole and WG Naming have dealt with a number of issues and comments from trolls. We anticipate that anyone using this document to guide their own work will receive the same kind of attention. 
```